### ISSUE TITLE
Petition to switch to old-ani-cli

### ISSUE THREAD

**Issue by Dink4n** [pystardust/ani-cli#264]
Title says it all

**Comment by ura43** [comment]
Yes, it does, but how?
There haven't just been 15 useless features I naively added over the past few days, but also a good chunk of bugfixes, mainly regarding parsing, authored by @RaynardGerraldo.
Let's also work out a list of features.
Nothing set in stone, but a document/wiki page we can reference when closing feature requests, as well as something to keep us grounded.

While this is a chunk of work, I support that move

**Comment by ura43** [comment]
Can't assign @RaynardGerraldo yet for some reason. I think only commenters can be assigned

**Comment by Dink4n** [comment]
I think you can only assign collaborators

**Comment by ura43** [comment]
No, thats not it, I assinged stuff to Ray before.

**Comment by ura43** [comment]
Yeah, assigning is stupid

**Comment by Dink4n** [comment]
I actually unassigned by mistake

**Comment by Dink4n** [comment]
Yeah, assigning is dumb

**Comment by ura43** [comment]
:rofl: 

Actually, why just switch to old-ani-cli instead of opening a branch "new-ani-cli" where we implement the stuff new? Like one giant refactor?

**Comment by Dink4n** [comment]
old-ani-cli is the "minimal" ani-cli. It's got all the "main" features, and the scraping implementation is the same as the current

**Comment by ura43** [comment]
I cloned old-ani-cli and the scraping doesn't work

**Comment by Dink4n** [comment]
Gogoanime changed the url

**Comment by Dink4n** [comment]
You can use [my fork](https://github.com/dink4n/ani-cli) which is a fork of old-ani-cli

**Comment by RaynardGerraldo** [comment]
Do we really need to go as far back as old-ani-cli tho? Is there no a slightly newer version than that, maybe @Dink4n 's branch?

**Comment by ura43** [comment]
What do we do about windows and mac?

**Comment by RaynardGerraldo** [comment]
> Do we really need to go as far back as old-ani-cli tho? Is there no a slightly newer version than that, maybe @Dink4n 's branch?



> You can use [my fork](https://github.com/dink4n/ani-cli) which is a fork of old-ani-cli

Does playing and downloading work there?

**Comment by ura43** [comment]
And even old ani cli isn't without flaws, but we can work on those.
The feature scope is a lot better

**Comment by Dink4n** [comment]
> Does playing and downloading work there?

Playing works hasn't downloaded anything in a long time.

**Comment by ura43** [comment]
Downloading works

**Comment by Dink4n** [comment]
How about going back to this commit 7b5ad5cd2ff1698b0a8372d2eddd86a62fc31384 in ani-cli

**Comment by RaynardGerraldo** [comment]
> What do we do about windows and mac?

Was there ever an older version of the windows branch? The last commit was last month tho so its probably broken by now ,did it use the same implementation of the main branch? If it does then we can just copy whatever main branch version we are going to use later.

And for mac the bug fix was changing setsid to nohup right?

**Comment by ura43** [comment]
Supporting mac is easy.
Is windows power shell posix compliant?

**Comment by RaynardGerraldo** [comment]
Wow downloading and playing works? Wonder why it broke in this version then

**Comment by Dink4n** [comment]
> Is windows power shell posix compliant?
No, it's not

**Comment by Dink4n** [comment]
> Was there ever an older version of the windows branch? The last commit was last month tho so its probably broken by now ,did it use the same implementation of the main branch? If it does then we can just copy whatever main branch version we are going to use later.

The only difference in the Windows branch is that it uses vlc as the default player.

**Comment by ura43** [comment]
I'd say the following would be an appealing minimal feature set
```
ani-cli <query>
-h help
-v version
-d download
-r resume / -c continue

(interactive)
[n] next
[s] select
[q] exit 
```
A query would always need to be passed as a parameter.
The histfile would be nothing more than sth to enable -r / -c.
Only official mpv support, whoever wants to use sth else should edit the script.

**Comment by Dink4n** [comment]
And we can release patches like [suckless](https://suckless.org/philosophy/)

**Comment by CoolnsX** [comment]
> I'd say the following would be an appealing minimal feature set
> 
> ```
> ani-cli <query>
> -c continue
> -d download
> -h help
> -v version
> 
> (interactive)
> [n] next
> [s] select
> [q] exit 
> ```
> 
> A query would always need to be passed as a parameter. The histfile would be nothing more than sth to enable -c. Only official mpv support, whoever wants to use sth else should edit the script.

Looks pretty good to me... And I never liked that history feature.. because never used it😂😂

**Comment by ura43** [comment]
Like how many times have you replayed an episode or even played the one before? 
select would be enough for those edge cases.
Even the history feature is pushing it a bit in terms of what is a minimal feature set

**Comment by Dink4n** [comment]
> Even the history feature is pushing it a bit in terms of what is a minimal feature set

We don't need to go that minimal

**Comment by ura43** [comment]
> And we can release patches like [suckless](https://suckless.org/philosophy/)

Bigbrain. Let's make terminal loving linux users our target audience again instead of trying to appeal to every pirate

**Comment by Dink4n** [comment]
That's a great idea!

**Comment by ura43** [comment]
> > Even the history feature is pushing it a bit in terms of what is a minimal feature set
> 
> We don't need to go that minimal

Yeah, I use the history so we keep it. Maybe make history work like current -u (unwatched) where you select what anime you want to continue watching, idk

**Comment by Dink4n** [comment]
I would like to keep the history as it works now.

**Comment by ura43** [comment]
Like if you don't use a tiling wm, don't use our program.
I'm exaggerating, but you get the point.
[animdl](https://github.com/justfoolingaround/animdl) can have the windows and mac users

**Comment by ura43** [comment]
> I would like to keep the history as it works now.

Then we have our first patch file :wink:

**Comment by Dink4n** [comment]
I have a port of [ani-cli in c](https://github.com/dink4n/canime). We can go suckless there.

**Comment by ura43** [comment]
No. Writing software that interacts with network services in C is unproductive.
Lets stay with posix shell. A good middle ground between python and c for our purposes

**Comment by Dink4n** [comment]
Fair enough

**Comment by ura43** [comment]
Just look how many files are in that repo of yours. We just need one shellscript and a readme

**Comment by Dink4n** [comment]
It's dependency free. I plan to remove the `curl` dependency

**Comment by Dink4n** [comment]
Ok, back to the main topic. We are talking about switching to old-ani-cli

**Comment by ura43** [comment]
> Ok, back to the main topic. We are talking about switching to old-ani-cli

**tldr: let's do it**

**Comment by ura43** [comment]
I already created the "bloated-ani-cli" branch from current master

**Comment by ura43** [comment]
While we're at it, there seem to be some residual branches we may wanna get rid of:
- dub option
- qa
- save-resume
- add-license-1

remaining would be:
- master (formerly old-ani-cli)
- windows-vlc
- bloated-ani-cli

**Comment by Dink4n** [comment]
Hey, can we go back to this commit 7b5ad5cd2ff1698b0a8372d2eddd86a62fc31384 instead of using old-ani-cli

Scraping doesn't work in that commit

**Comment by CoolnsX** [comment]
Okay but remove the extra bloat(features) after that..

**Comment by ura43** [comment]
I say we just straightup move to old ani-cli
Our Todo's after the move:
@ura43 
- Write readme
- Helptext refactor, TUI refactor
- Shellcheck action

@RaynardGerraldo 
- Backport parsing bug fixes

@Dink4n 
- Refactoring of helper functions

@CoolnsX
- Code review us

Does that seem about right?

**Comment by CoolnsX** [comment]
Okay.. and I can do that much right now(because of exams, i have tomorrow)... Okay I am in..

**Comment by ura43** [comment]
Well, we can do the stuff tomorrow if you have no time today, or just review each other.
And you refactor our refactors tomorrow.
I have nothing planned for the next few hours.
How are you available @RaynardGerraldo @Dink4n ?

**Comment by Dink4n** [comment]
There's not much to do in old-ani-cli. I have time tomorrow.

**Comment by ura43** [comment]
Shouldn't take longer than an hour for each of us

**Comment by RaynardGerraldo** [comment]
> Well, we can do the stuff tomorrow if you have no time today, or just review each other. And you refactor our refactors tomorrow. I have nothing planned for the next few hours. How are you available @RaynardGerraldo @Dink4n ?



> Well, we can ~do the stuff tomorrow if you have no time today, or~ just review each other. And you refactor our refactors tomorrow. I have nothing planned for the next few hours. How are you available @RaynardGerraldo @Dink4n ?

Yea im in, i'll try and do it tomorrow my time. I have a lot to do today ,just read through the discussion this seems promising

**Comment by tekmyndaspy** [comment]
I feel the benefits to come from switching to the less bloated and more focused old-ani-cli would outweigh the work to come from reintroducing bugfixes, especially as the bug fixes _are_ kept in the branch, so one would be able to see what has been done to do it again. It may take some work, but I feel a bit of extra work is worth it to avoid code spaghettification. If I had much experience in scripting, I would do what I could to help, but unfortunately I lack the knowledge.

**Comment by ArekkusuMagni** [comment]
> > And we can release patches like [suckless](https://suckless.org/philosophy/)
> 
> Bigbrain. Let's make terminal loving linux users our target audience again instead of trying to appeal to every pirate

That is who it should have been this whole time. I am a linux noob but even I understand you can't appeal to everyone.

**Comment by CoolnsX** [comment]
> I feel the benefits to come from switching to the less bloated and more focused old-ani-cli would outweigh the work to come from reintroducing bugfixes, especially as the bug fixes _are_ kept in the branch, so one would be able to see what has been done to do it again. It may take some work, but I feel a bit of extra work is worth it to avoid code spaghettification. If I had much experience in scripting, I would do what I could to help, but unfortunately I lack the knowledge.

Yeah... Linux is all about simplicity.. that's why most linux user use Arch Linux because of its simplicity...
And complex code will introduce more bugs.. and to fix it requires more complex code.. which in turn increases complexity... And cycle continues...

**Comment by 0x8FF** [comment]
What. Isn't the old branch broken?

**Comment by 0x8FF** [comment]
Can we all talk about this on Discord/Matrix/IRC Please

**Comment by 0x8FF** [comment]
Also may I ask, what do you consider bloat in this project? The fact that we provide support with windows gitbash? Is it because we provide the offer of using MPV or VLC?

**Comment by Dink4n** [comment]
We have 3 choices now

* Fix the current implementation
* Switch to old-ani-cli
* Switch to this commit 7b5ad5cd2ff1698b0a8372d2eddd86a62fc31384 or Switch to windows-vlc branch

After you guys decide what to do, ping me.

**Comment by RaynardGerraldo** [comment]
> We have 3 choices now
> 
> * Fix the current implementation
> * Switch to old-ani-cli
> * Switch to this commit [7b5ad5c](https://github.com/pystardust/ani-cli/commit/7b5ad5cd2ff1698b0a8372d2eddd86a62fc31384) or Switch to windows-vlc branch
> 
> After you guys decide what to do, ping me.

Just to clarify,both commit [7b5ad5c] and old-ani-cli(im assuming ur branch) works right? Or is there something broken that i should know of,we could just take the working parts from either code and implement it to the simplest version we have right now, old-ani-cli(?)

**Comment by Dink4n** [comment]
Scraping doesn't work in 7b5ad5c. It's pretty easy to fix.

**Comment by RaynardGerraldo** [comment]
> Scraping doesn't work in [7b5ad5c](https://github.com/pystardust/ani-cli/commit/7b5ad5cd2ff1698b0a8372d2eddd86a62fc31384). It's pretty easy to fix.

Yea playing and downloading doesnt work here ,i think i implemented a fix right after that commit, but your branch is working fine so why not use that?

**Comment by CoolnsX** [comment]
Do we need video quality option???..
If no, then we can remove video_quality function and make the get_links() scrape only download links.. instead of scraping links...

**Comment by ura43** [comment]
> Can we all talk about this on Discord/Matrix/IRC Please

My matrix is on my profile

**Comment by ura43** [comment]
I'd suggest we fix the current implementation.
I know yesterday I was for just switching to old-ani-cli, but many edge cases have been adressed and it generally has better error handling

**Comment by 0x8FF** [comment]
^
